src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/GetTaskByIdRolePermissionsTest.java

```diff
@@ -0,0 +1,111 @@
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_DUE_DATE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_HAS_WARNINGS;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_NAME;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_TITLE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_TYPE;
+
+public class GetTaskByIdRolePermissionsTest extends SpringBootFunctionalBaseTest {
+
+    private static final String TASK_INITIATION_ENDPOINT_BEING_TESTED = "task/{task-id}";
+    private static final String ENDPOINT_BEING_TESTED = "task/{task-id}/roles";
+    private Headers authenticationHeaders;
+
+    @Before
+    public void setUp() {
+        authenticationHeaders = authorizationHeadersProvider.getTribunalCaseworkerAAuthorization("wa-ft-test-r2-");
+    }
+
+    @Test
+    public void should_return_200_and_retrieve_role_permission_information_for_given_task_id() {
+        TestVariables taskVariables = common.setupTaskAndRetrieveIds();
+        String taskId = taskVariables.getTaskId();
+        common.setupOrganisationalRoleAssignment(authenticationHeaders);
+
+        initiateTask(taskVariables);
+
+        Response result = restApiActions.get(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            authenticationHeaders
+        );
+
+        result.then().assertThat()
+            .statusCode(HttpStatus.OK.value())
+            .contentType(APPLICATION_JSON_VALUE)
+            .body("roles.size()", equalTo(4))
+            .body("roles[0].role_category", is("LEGAL_OPERATIONS"))
+            .body("roles[0].role_name", is("case-manager"))
+            .body("roles[0].permissions", hasItems("Read","Own","Refer"))
+            .body("roles[0].authorisations", empty())
+            .body("roles[1].role_category", equalTo("LEGAL_OPERATIONS"))
+            .body("roles[1].role_name", is("senior-tribunal-caseworker"))
+            .body("roles[1].permissions", hasItems("Read","Own","Refer"))
+            .body("roles[1].authorisations", empty())
+            .body("roles[2].role_category", nullValue())
+            .body("roles[2].role_name", is("task-supervisor"))
+            .body("roles[2].permissions", hasItems("Read","Cancel","Manage","Refer"))
+            .body("roles[2].authorisations", empty())
+            .body("roles[3].role_category", is("LEGAL_OPERATIONS"))
+            .body("roles[3].role_name", is("tribunal-caseworker"))
+            .body("roles[3].permissions", hasItems("Read","Own","Refer"))
+            .body("roles[3].authorisations", empty());
+
+        common.cleanUpTask(taskId);
+    }
+
+    private void initiateTask(TestVariables testVariables) {
```

Comment:
Is this type of method not in a common class already somewhere?

Reply:
I didn't find but will add it in RWA-810

Reply:
Unfortunately not, can have a tech debt to move onto common class.

Reply:
Added to common class in RWA-810.

Reply:
ok